indexer/services/comlink/src/controllers/api/v4/addresses-controller.ts

```diff
@@ -241,6 +241,107 @@ class AddressesController extends Controller {
     );
     return subaccountResponse;
   }
+
+  @Get('/:address/parentSubaccountNumber/:parentSubaccountNumber')
+  public async getParentSubaccount(
+    @Path() address: string,
+      @Path() parentSubaccountNumber: number,
+  ): Promise<ParentSubaccountResponse> {
+
+    const childSubaccountIds: string[] = getChildSubaccountNums(parentSubaccountNumber).map(
+      (subaccountNumber: number): string => SubaccountTable.uuid(address, subaccountNumber),
+    );
+
+    // TODO(IND-189): Use a transaction across all the DB queries
+    const [subaccounts, latestBlock]: [
+      SubaccountFromDatabase[],
+      BlockFromDatabase,
+    ] = await Promise.all([
+      SubaccountTable.findAll(
+        {
+          id: childSubaccountIds,
+          address,
+        },
+        [],
+      ),
+      BlockTable.getLatest(),
+    ]);
+
+    if (subaccounts.length === 0) {
+      throw new NotFoundError(`No subaccounts found for address ${address} and parentSubaccountNumber ${parentSubaccountNumber}`);
+    }
+
+    const latestFundingIndexMap: FundingIndexMap = await FundingIndexUpdatesTable
+      .findFundingIndexMap(
+        latestBlock.blockHeight,
+      );
+
+    const subaccountResponses: SubaccountResponseObject[] = await Promise.all(subaccounts.map(
+      async (subaccount: SubaccountFromDatabase): Promise<SubaccountResponseObject> => {
+        const [
+          perpetualPositions,
+          assetPositions,
+          assets,
+          markets,
+          lastUpdatedFundingIndexMap,
+        ] = await Promise.all([
+          getOpenPerpetualPositionsForSubaccount(
+            subaccount.id,
+          ),
+          getAssetPositionsForSubaccount(
+            subaccount.id,
+          ),
+          AssetTable.findAll(
+            {},
+            [],
+          ),
+          MarketTable.findAll(
+            {},
+            [],
+          ),
```

Comment:
we don't need to pull this information for every subaccount, this only needs to be done once